spec/ddtrace/contrib/rest_client/request_patch_spec.rb

```diff
@@ -0,0 +1,170 @@
+      it 'creates a span' do
+        expect { request }.to change { tracer.writer.spans.first }.to be_instance_of(Datadog::Span)
+      end
+
+      it 'returns response' do
+        expect(request.body).to eq(response)
+      end
+
+      describe 'created span' do
+        subject(:span) { tracer.writer.spans.first }
+
+        context 'response is successfull' do
+          before do
+            request
+          end
+
+          it 'has tag with target host' do
+            expect(span.get_tag(Datadog::Ext::NET::TARGET_HOST)).to eq(host)
+          end
+
+          it 'has tag with target port' do
+            expect(span.get_tag(Datadog::Ext::NET::TARGET_PORT)).to eq('80')
+          end
+
+          it 'has tag with target port' do
+            expect(span.get_tag(Datadog::Ext::HTTP::METHOD)).to eq('GET')
+          end
+
+          it 'has tag with target port' do
+            expect(span.get_tag(Datadog::Ext::HTTP::URL)).to eq(path)
+          end
+
+          it 'has tag with status code' do
+            expect(span.get_tag(Datadog::Ext::HTTP::STATUS_CODE)).to eq(status.to_s)
+          end
+
+          it 'is http type' do
+            expect(span.span_type).to eq('http')
+          end
+
+          it 'is named correctly' do
+            expect(span.name).to eq('rest_client.request')
+          end
+        end
+
+        context 'response has internal server error status' do
+          let(:status) { 500 }
+
+          before do
+            expect { request }.to raise_exception(RestClient::InternalServerError)
+          end
+
+          it 'has tag with status code' do
+            expect(span.get_tag(Datadog::Ext::HTTP::STATUS_CODE)).to eq(status.to_s)
+          end
+
+          it 'has error set' do
+            expect(span.get_tag(Datadog::Ext::Errors::MSG)).to eq('500 Internal Server Error')
+          end
+          it 'has error stack' do
+            expect(span.get_tag(Datadog::Ext::Errors::STACK)).not_to be_nil
+          end
+          it 'has error set' do
+            expect(span.get_tag(Datadog::Ext::Errors::TYPE)).to eq('RestClient::InternalServerError')
+          end
+        end
+
+        context 'response has not found status' do
+          let(:status) { 404 }
+
+          before do
+            expect { request }.to raise_exception(RestClient::NotFound)
+          end
+
+          it 'has tag with status code' do
+            expect(span.get_tag(Datadog::Ext::HTTP::STATUS_CODE)).to eq(status.to_s)
+          end
+
+          it 'error is not set' do
+            expect(span.get_tag(Datadog::Ext::Errors::MSG)).to be_nil
+          end
+        end
+      end
+    end
+
+    it_behaves_like 'instrumented request'
+
+    context 'distrubuted tracing enabled' do
+      let(:rest_client_options) { { distributed_tracing: true } }
+
+      it_behaves_like 'instrumented request'
+
+      it 'propagates the headers' do
+        request
+
+        span = tracer.writer.spans.first
```

Comment:
You could bump this `span =` to a lazy `let(:span)` block, which would not be evaluated until you reference it, allowing you to reduce repetition.

Reply:
done